### PR TITLE
Abi | add. error message when api fails

### PR DESCRIPTION
**Pull request by abinaya-u**
Asana: https://app.asana.com/0/1204322126657135/1207086955567548/f

### CODE REVIEW

File: src/context/AllMedications.jsx [hunk 1]
```diff
@@ -22,8 +22,7 @@ export const AllMedicationsContextProvider = (props) => {
       }));
     } catch (error) {
       setMedications((prevState) => ({
-        ...prevState,
-        isLoading: false,
+        isLoading: true,
```
Comment (ayush34957) [inline src/context/AllMedications.jsx:25]: why we are making isLoading as true in catch block as when we call this method we make this as true initially and based on success and failure we set it as false

File: src/features/DisplayControls/DrugChart/components/DrugChartView.jsx [hunk 2]
```diff
@@ -87,15 +90,23 @@ export default function DrugChartWrapper(props) {
...
-      setIsShiftButtonsDisabled({
-        previous:
-          (isReadMode && response.data.length === 0) ||
-          response.data[0].startDate > startDateTimeInSeconds,
-        next:
-          startDateTimeInSeconds >= nextShiftMaxHour ||
-          endDateTimeInSeconds >= nextShiftMaxHour,
-      });
+      if (response?.error) {
+        if (response.error.response.status === 403) {
```
Comment (ayush34957) [inline src/features/DisplayControls/DrugChart/components/DrugChartView.jsx:94]: can we move this status code to a src/constant file and use that from there?

File: src/features/DisplayControls/DrugChart/components/DrugChartView.jsx [hunk 3]
```diff
@@ -304,7 +315,9 @@ export default function DrugChartWrapper(props) {
           <Loading withOverlay={false} />
         </div>
       ) : transformedData && transformedData.length === 0 ? (
-        <div className="no-nursing-tasks">{NoMedicationTaskMessage}</div>
+        <div className="no-nursing-tasks">
+          {errorMessage !== "" ? errorMessage : NoMedicationTaskMessage}
```
Comment (ayush34957) [inline src/features/DisplayControls/DrugChart/components/DrugChartView.jsx:319]: we can also write the above code as:
errorMessage ? errorMessage : NoMedicationTaskMessage
The above would also handle null and undefined scenarios

File: src/features/DisplayControls/NursingTasks/components/NursingTasks.jsx [hunk 4]
```diff
@@ -239,60 +245,69 @@ export default function NursingTasks(props) {
...
-          isReadMode,
-          excludePRN
-        );
-        setMedicationNursingTasks(
-          extractedNonMedicationTasks.length > 0
-            ? [...extractedData, ...extractedNonMedicationTasks]
-            : extractedData
-        );
+    if (nursingTasks?.error) {
+      if (nursingTasks.error.response.status === 403) {
```
Comment (ayush34957) [inline src/features/DisplayControls/NursingTasks/components/NursingTasks.jsx:249]: same for all files

File: src/features/DisplayControls/Treatments/components/Treatments.jsx [hunk 5]
```diff
@@ -25,7 +25,7 @@ import {
   getSlotsForAnOrderAndServiceType,
 } from "../utils/TreatmentsUtils";
 import { getCookies } from "../../../../utils/CommonUtils";
-import { defaultDateTimeFormat, serviceType } from "../../../../constants";
+import { ForbiddenErrorMessage, GenericErrorMessage, defaultDateTimeFormat, serviceType } from "../../../../constants";
```
Comment (ayush34957) [inline src/features/DisplayControls/Treatments/components/Treatments.jsx:28]: check here for prettier code format in this file?

File: src/context/AllMedications.jsx (hunk 1 above)
Comment (abinaya-u) [reply src/context/AllMedications.jsx]: this change was unintentional, changed it back to false

File: src/features/DisplayControls/DrugChart/components/DrugChartView.jsx (hunk 2 above)
Comment (abinaya-u) [reply src/features/DisplayControls/DrugChart/components/DrugChartView.jsx]: extracted to a constant

File: src/features/DisplayControls/DrugChart/components/DrugChartView.jsx (hunk 3 above)
Comment (abinaya-u) [reply src/features/DisplayControls/DrugChart/components/DrugChartView.jsx]: changes have been made

**Review by ayush34957 — APPROVED** [review]

**Pull request merged**